src/components/Articles.jsx

```diff
@@ -8,27 +8,52 @@ function Articles({isLoading, setIsLoading}) {
     const { topic } = useParams();
     const [articles, setArticles] = useState([]);
     const [mostRecent, setMostRecent] = useState({});
+    const [query, setQuery] = useState("?sort_by=created_at&order=desc");
+    const [errorMessage, setErrorMessage] = useState("");
 
     useEffect(() => {
+        setErrorMessage("");
         setIsLoading(true);
-        getArticles(topic).then((articlesFromApi) => {
+        getArticles(topic, query).then((articlesFromApi) => {
           setArticles(articlesFromApi);
           setMostRecent(articlesFromApi[0]);
           setIsLoading(false);
+        }).catch((err) => {
+          setErrorMessage("Oops, something went wrong loading the articles! Please try again.")
         })
-      }, [setIsLoading, topic])
+      }, [setIsLoading, topic, query])
+
+      const handleQuery = (e) => {
+        e.preventDefault();
+        setQuery(e.target.value);
+      }
 
     if (isLoading) {
         return (
             <img src="https://img.pikbest.com/png-images/20190918/cartoon-snail-loading-loading-gif-animation_2734139.png!bw700" alt="loading" id="loading-img"/>
         )
     }
 
+    if (errorMessage !== "") {
+      return <p>{errorMessage}</p>
+    }
+
     else {
 
     return (
       <section className="articles-section">
-        <MostRecent mostRecent={mostRecent}/>
+        <form id="sort-by-form">
+          <label htmlFor="sort-by">Sort by: </label>
+          <select onChange={handleQuery} id="sort-by" value={query}>
+            <option value="?sort_by=created_at&order=desc">Date: Newest first</option>
```

Comment:
This is fair enough but separate buttons for sort-by and order might be a nicer UX (perfectly fine to leave it as it is, though!)